content/blog/2018/06/2018-06-19-jenkins-java10-hackathon-day2.adoc

```diff
@@ -0,0 +1,117 @@
+---
+
+image:/images/logos/formal_java/256.png[Jenkins Java, role=center, float=right]
+
+As you probably know, this week we have a
+link:/blog/2018/06/08/jenkins-java10-hackathon/[Jenkins & Java 10 Online Hackathon].
+This is an open online event, where we work together on Jenkins core and plugins in order
+to find and fix compatibility issues, share experiences and have some fun.
+Everybody is welcome to join, independently of their Jenkins experience and amount of time they have available.
+
+After the link:https://youtu.be/1HrgWs0l0e8[kicked off] on Monday
+Jenkins contributors have been working on Java 10 and Java 11 support in Jenkins.
+We have already received contributions from 12 hackathon participants, and the number keeps counting.
+There are still 3 days ahead, but we have already achieved some important results we want to share.
+
+### Jenkins Pipeline
+
+One of our major efforts over last 2 days was to get Jenkins Pipeline working on
+Java 10+.
+When the hackathon started Jenkins Pipeline was not working at all,
+and it was a major blocker for Java support and for exploratory testing in particular.
+We've been working together with Sam van Oort and Devin Nusbaum to fix the libraries in
+the Jenkins core, Pipeline: Support plugin and Docker packaging.
+
+Just to summarize the result of two days in one screenshot...
+
+image:/images/post-images/2018-06-19-java10-hackathon-day-2/successful-pipeline.png[Successful Pipeline on Java 10, role=center]
+
+Yes, we have got it running!
+Over two days we have got from the "Pipeline Crashes Immediately"
+state to the situation when the most of key Pipeline features are operational,
+including Scripted and Declarative Pipeline, BlueOcean, shared libraries and
+dozens of plugins being used in the Jenkins plugin build flow.
+
+There is still a lot of work to do to get the changes finalized,
+but Jenkins Pipeline is available for testing on Java 10 and 11 now.
+If you want to try it out, you can use a new `jenkins/jenkins-experimental:blueocean-jdk10`
+image we have created.
+It bundles all the required patches, so you can just run the following command to get started:
+
+```
+docker run -p 8080:8080 -p 50000:50000 jenkins/jenkins-experimental:blueocean-jdk10
+```
+
+If you want to try more complex scenarions, see the
+link:/blog/2018/06/17/running-jenkins-with-java10-11/[Running Jenkins with Java 10 and 11 blogpost]
+and link:https://docs.google.com/document/d/1ed6wFOlq4cWrSL6UkCSzFbaY80AT-sk8ncB4Fz5QXyM/edit#heading=h.8lwu94cr28ig[List of Required patches].
+
+### What else?
+
+Although Pipeline is the most visible change,
+there are other ongoing activities:
+
+* Devin Nusbaum explored plugin startup issues we had with JDK 11ea+17
+  and confirmed that we need to upgrade our images to JDK 11ea+18
+* Gianpaolo Macario is working on adopting the Java 10 images in his easy-jenkins project
+** https://github.com/gmacario/easy-jenkins/pull/266
+* Sam van Oort are working on Jenkins development tools
```

Comment:
🐛 I'm actually working on just getting tests and normal compilation working (using Java 10 with Maven).   Devin is doing multi-release JARs.

Reply:
Unless there is a specific need for multi-release JARs in pipeline, I think it makes sense to centralize effort around https://github.com/ndeloof/fields which should cover the most common use cases. I can look into what it will take to get these working, but I think Nicolas is already working on that with getting Stapler and all of its dependents updated to use the library.